ZenPacks/zenoss/OpenStack/tests/test_modeler.py

```diff
@@ -0,0 +1,127 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2016, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import os
+import logging
+import json
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger('zen.OpenStack')
+
+from zope.event import notify
+
+import Globals
+
+from Products.Zuul.catalog.events import IndexingEvent
+from Products.Five import zcml
+
+from Products.DataCollector.ApplyDataMap import ApplyDataMap
+from Products.ZenTestCase.BaseTestCase import BaseTestCase
+from Products.ZenUtils.Utils import unused
+
+import ZenPacks.zenoss.OpenStack
+from ZenPacks.zenoss.OpenStack.tests.utils import setup_crochet
+from ZenPacks.zenoss.OpenStack.modeler.plugins.zenoss.OpenStack \
+import OpenStack as OpenStackModeler
+
+from utils import DictToObject
+
+
+unused(Globals)
+crochet = setup_crochet()
+
+
+class TestModel(BaseTestCase):
+
+    disableLogging = False
+
+    def afterSetUp(self):
+        super(TestModel, self).afterSetUp()
+
+        dc = self.dmd.Devices.createOrganizer('/Devices/OpenStack')
+
+        dc.setZenProperty('zPythonClass', 'ZenPacks.zenoss.OpenStack.Endpoint')
+        dc.setZenProperty('zOpenStackAuthUrl', '')
+        dc.setZenProperty('zOpenstackComputeApiVersion', [])
+        dc.setZenProperty('zOpenStackProjectId', [])
+        dc.setZenProperty('zOpenStackInsecure', [])
+        dc.setZenProperty('zOpenStackRegionName', 'RegionOne')
+
+        self.d = dc.createInstance('zenoss.OpenStack.testDevice')
+        self.d.setPerformanceMonitor('localhost')
+        self.d.index_object()
+
+        notify(IndexingEvent(self.d))
+        self.applyDataMap = ApplyDataMap()._applyDataMap
+
+        zcml.load_config('configure.zcml', ZenPacks.zenoss.OpenStack)
+
+        self._loadZenossData()
+
+    def _loadZenossData(self):
+        if hasattr(self, '_modeled'):
+            return
+
+        modeler = OpenStackModeler()
+        with open(os.path.join(os.path.dirname(__file__),
+                               'data',
+                               'modeldata.json')) as json_file:
+            results = json.load(json_file)
+
+        results = self._dict_to_object(results)
+
+        for data_map in modeler.process(self.d, results, log):
+            self.applyDataMap(self.d, data_map)
+        self._modeled = True
+
+    def _dict_to_object(self, results):
```

Comment:
move the method to utils.py, use one style within the module, in this case I'm talking about the method's name